File: tag/Array/001. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
    	vector<int> ret;
        if(nums.empty()) return ret;
        vector<int> hash(100000, -1);
        for(int i = 0; i < nums.size(); i++){
        	hash[target - nums[i] + 50000] = i;
        }
        for(int i = 0; i < nums.size(); i++){
        	if(hash[nums[i] + 50000] >= 0 && hash[nums[i] + 50000] != i){
        		ret.push_back(i);
        		ret.push_back(hash[nums[i] + 50000]);
        		break;
        	}
        }
        return ret;

    }
};

File: java/Array/001. Two Sum.java
import java.util.HashMap;
import java.util.Map;

public class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hashMap = new HashMap<>();
        for(int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if(hashMap.containsKey(complement)) {
                int ret[] = {hashMap.get(complement), i};
                return ret;
            }
            hashMap.put(nums[i], i);
        }
        throw new IllegalArgumentException("No Result");
    }

    public static void main(String[] args){}
}
